### DISCUSSION TITLE
Which specfic file is the "added_tokens.json" file required for GPT4ALL?

### DISCUSSION THREAD

**Discussion by Dwedit** [ggml-org/llama.cpp#1245] [Q&A]
What is the "added_tokens.json" file required to use GPT4ALL?  The instructions say to use "one from Alpaca".  But there are many many different projects based on Alpaca which have that in the name, and many different variations of "added_tokens.json".  I am seeing mostly blank versions of such a file whenever I look for one.  I have no idea which file I actually need.  Can someone provide the exact file, or at least the SHA256 hash of the necessary file?

For reference, my SHA-256 of GPT4ALL's model is "05c9dc0a4904f3b232cffe717091b0b0a8246f49c3f253208fbf342ed79a6122 *gpt4all-lora-quantized.bin"

**Comment by quantonganh** [comment]
I tried [this](https://huggingface.co/chavinlo/alpaca-13b/blob/main/added_tokens.json) one, and `gpt4all-lora-quantized.bin` could be converted to `ggml-model-q4_0.bin`, but it failed to load:

```sh
$ ./main -m ./models/gpt4all-7B/ggml-model-q4_0.bin -t 8 -n 512 -p 'send http request in golang'
main: build = 917 (1a94186)
main: seed  = 1690515134
llama.cpp: loading model from ./models/gpt4all-7B/ggml-model-q4_0.bin
error loading model: unexpectedly reached end of file
llama_load_model_from_file: failed to load model
llama_init_from_gpt_params: error: failed to load model './models/gpt4all-7B/ggml-model-q4_0.bin'
main: error: unable to load model
```
